server/templates/details.html.tmpl

```diff
@@ -76,3 +97,74 @@
   {{end}}
   <p class="text-dark-gray3 text-sm">{{or .Error .StatusDescription}}</p>
 {{end}}
+
+{{define "result-predicates-details"}}
+  {{ $s := (or (and .Error "error") (.Status | print)) }}
+  {{if ne $s "skipped"}}
+    <b class="font-bold text-sm">This Rule is Selected Due to</b>
+  {{else}}
+    <b class="font-bold text-sm">This Rule is Skipped Due to</b>
+  {{end}}
+  {{template "result-predicates-info" .}}
+{{end}}
+
+{{define "result-predicates-info"}}
+  {{ $s := (or (and .Error "error") (.Status | print)) }}
+  <ul class="list-decimal list-inside">
+  {{range .PredicatesInfo}}
+    {{$predicateData := getPredicateInfo .}}
+    <li class="pt-2 text-sm">
+        {{if ne $s "skipped"}}
+          Predicate <span class="bg-light-gray1 font-bold">{{.Name}}</span> is satisfied because the {{$predicateData.PredicateDataName}} :
+        {{else}}
+          Predicate <span class="bg-light-gray1 font-bold">{{.Name}}</span> is not satisfied because the {{$predicateData.PredicateDataName}} :
+        {{end}}
+        <ul class="list-disc list-inside">
+          {{range $predicateData.PredicateValues}}
+            <li class="ml-4">
+                {{.}}
+            </li>
+          {{end}}
+          {{if not $predicateData.PredicateValues}}
+             <span class="pl-2">-</span>
+          {{end}}
+        </ul>
+        {{if ne $s "skipped"}}
+          match
+        {{else}}
+          do not match
+        {{end}}
+        <ul class="list-none list-inside">
+          {{range $key, $value := getPredicateRequirement .}}
+            <li class="ml-4">
+              <p>{{$key}}</p>
+              <ul class="list-disc list-inside">
+                {{range $value}}
+                  <li class="ml-4">{{.}}</li>
+                {{end}}
+              </ul>
+            </li>
+          {{end}}
+        </ul>
+    </li>
+  {{end}}
+  </ul>
+{{end}}
+
+{{define "result-requires-details"}}
+  <b class="font-bold text-sm">This Rule Requires Review from:</b>
```

Comment:
This might not be well-suited to for the disapproval case which also uses this template. It's kind of weird to say a disapproval rule requires approval from

Reply:
That's fair, let's leave it as "requires review" for now.